src/linear_algebra/bilinear_form.lean

```diff
@@ -136,8 +151,57 @@ def bilin_linear_map_equiv : (bilin_form R₂ M) ≃ₗ[R₂] (M →ₗ[R₂] M
   left_inv := λ B, by {ext, refl},
   right_inv := λ B, by {ext, refl} }
 
+@[norm_cast]
+lemma coe_fn_to_linear_map (x : M) : ⇑(F.to_linear_map x) = F x := rfl
+
+lemma map_sum_left {α} (B : bilin_form R₂ M) (t : finset α) (g : α → M) (w : M) :
+  B (t.sum g) w = t.sum (λ i, B (g i) w) :=
+show B.to_linear_map (t.sum g) w = t.sum (λ i, B (g i) w),
+by { rw [B.to_linear_map.map_sum, linear_map.coe_fn_sum, finset.sum_apply], norm_cast }
+
+lemma map_sum_right {α} (B : bilin_form R₂ M) (t : finset α) (g : α → M) (v : M) :
+  B v (t.sum g) = t.sum (λ i, B v (g i)) :=
+(B.to_linear_map v).map_sum
+
 end
 
+section comp
+
+variables {N : Type w} [add_comm_group N] [module R N]
+
+/-- Apply a linear map on the left and right argument of a bilinear form. -/
+def comp (B : bilin_form R N) (l r : M →ₗ[R] N) : bilin_form R M :=
+{ bilin := λ x y, B (l x) (r y),
+  bilin_add_left := λ x y z, by simp [add_left],
+  bilin_smul_left := λ x y z, by simp [smul_left],
+  bilin_add_right := λ x y z, by simp [add_right],
+  bilin_smul_right := λ x y z, by simp [smul_right] }
+
+/-- Apply a linear map to the left argument of a bilinear form. -/
+def comp_left (B : bilin_form R M) (f : M →ₗ[R] M) : bilin_form R M :=
+B.comp f linear_map.id
+
+/-- Apply a linear map to the right argument of a bilinear form. -/
+def comp_right (B : bilin_form R M) (f : M →ₗ[R] M) : bilin_form R M :=
+B.comp linear_map.id f
+
+@[simp] lemma comp_left_comp_right (B : bilin_form R M) (l r : M →ₗ[R] M) :
+  (B.comp_left l).comp_right r = B.comp l r := rfl
+
+@[simp] lemma comp_right_comp_left (B : bilin_form R M) (l r : M →ₗ[R] M) :
+  (B.comp_right r).comp_left l = B.comp l r := rfl
+
+@[simp] lemma coe_fn_comp (B : bilin_form R N) (l r : M →ₗ[R] N) (v w) :
```

Comment:
In other parts of the library we seem to name these `comp_apply`. Of course here we are applying two arguments... so it is slightly different. But I would still think the same naming convention could be used. What do you think?

Reply:
The difference between "apply one argument" and "apply two arguments" doesn't seem too big, compared to "apply" and "coerce to a function". I'll change it :+1: